pkg/pillar/base/logobjecttypes.go

```diff
@@ -111,9 +111,14 @@ func NewLogObject(objType LogObjectType, objName string, objUUID uuid.UUID, key
 		log.Fatal("NewLogObject: objType and key parameters mandatory")
 	}
 	// Check if we already have an object with the given key
-	object, ok := logObjectMap[key]
+	var object *LogObject
+	value, ok := logObjectMap.Load(key)
 	if ok {
-		return object
+		object, ok = value.(*LogObject)
+		if ok {
+			return object
+		}
+		log.Fatal("NewLogObject: Object found in key map is not of type *LogObject")
```

Comment:
Print the actual type please. %T is the formatting directive for printing the type.

Reply:
done